### ISSUE TITLE
ARIA Presentational Roles Conflict Resolution

### ISSUE THREAD

**Issue by pkra** [validator/validator#2047]
### Summary

I haven't tested exhaustively and it may be out of scope but it seems authoring errors triggering conflict resolution (see https://w3c.github.io/aria/#conflict_resolution_presentation_none) are not flagged. 

### Is your feature request related to a problem?

Presentational roles conflict resolution is messy and authors are often not aware of it.

### Describe in more detail the solution you’d like

In particular, bullet 1 and 3 seem feasible, e.g. flagging `role="none/presentation"`with `tabindex` or with a [global ARIA attribute](https://w3c.github.io/aria/#dfn-global) as an author error. 

Ideally, validator could also do something for bullet 2 ("allowed children with explicit roles").

### Describe alternatives you’ve considered

I'm not sure what to write here.

### Use cases

Authors do things like 

```html
<nav role="none" aria-label="main"></nav>
<nav role="none" tabindex="0"></nav>
<svg role="none" aria-label="download"</svg>
```

and are then surprised that the role is ignored.

Currently, these examples do not raise any warnings or errors.

### Additional context

Happy to seek more real world examples, in particular for the second bullet point ("allowed children with explicit roles").

**Comment by sideshowbarker** [comment]
What we’d need it a specific set of tests.

**Comment by pkra** [comment]
Thanks so much for considering this.

The following tests might be a good starting point:

- https://wpt.fyi/results/wai-aria/role/role_none_conflict_resolution.html?label=master&label=experimental&aligned / https://github.com/web-platform-tests/wpt/blob/af55136b53/wai-aria/role/role_none_conflict_resolution.html
  -  maybe also these open PRs https://github.com/web-platform-tests/wpt/pull/50045 https://github.com/web-platform-tests/wpt/pull/50002 (but I didn't look through them carefully
- https://github.com/dequelabs/axe-core/blob/a4b809183f43c4296a3ec57cd80d8a8f34743361/test/integration/rules/presentation-role-conflict/presentation-role-conflict.html

These also don't cover a lot of the theoretical use cases but they have at least proven useful elsewhere.

**Comment by sideshowbarker** [comment]
Fixed in https://github.com/validator/validator/commit/2b143f477eb99caf877341039d6ac4ff00df647a and pushed to https://validator.w3.org/nu/

**Comment by pkra** [comment]
Thanks so much, @sideshowbarker!

**Comment by pkra** [comment]
@sideshowbarker testing this today, the following still passes on https://validator.w3.org/nu/

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width">
  <title>Test</title>
</head>
<body>
  <svg role="none" aria-label="download"></svg>
</body>
</html>
```

**Comment by sideshowbarker** [comment]
Fixed in https://github.com/validator/validator/commit/341edcd05bb4f9f1b2c1a168799041fecf5fbdea and deployed:

https://validator.w3.org/nu/?showsource=yes&doc=data%3Atext%2Fhtml%3Bbase64%2CPCFET0NUWVBFIGh0bWw%2BCjxodG1sIGxhbmc9ImVuIj4KPGhlYWQ%2BCiAgPG1ldGEgY2hhcnNldD0idXRmLTgiPgogIDxtZXRhIG5hbWU9InZpZXdwb3J0IiBjb250ZW50PSJ3aWR0aD1kZXZpY2Utd2lkdGgiPgogIDx0aXRsZT5UZXN0PC90aXRsZT4KPC9oZWFkPgo8Ym9keT4KICA8c3ZnIHJvbGU9Im5vbmUiIGFyaWEtbGFiZWw9ImRvd25sb2FkIj48L3N2Zz4KPC9ib2R5Pgo8L2h0bWw%2B

(P.S. I cases like this it’s probably better to open a new issue than to add a comment to a closed issue.)

**Comment by pkra** [comment]
@sideshowbarker thanks again! 

> I cases like this it’s probably better to open a new issue than to add a comment to a closed issue.)

Right. I had followed up here because I didn't have time to check a local installation and I wasn't sure if it was just the site having failed to update.
